src/omv/ports/stm32/omv_gpu.c

```diff
@@ -167,10 +167,14 @@ int omv_gpu_draw_image(image_t *src_img,
 
     #if __DCACHE_PRESENT
     // Ensures any cached writes to dst16 are flushed.
-    uint16_t *dst16_tmp = dst16;
-    for (int i = 0; i < dst_rect->h; i++) {
-        SCB_CleanInvalidateDCache_by_Addr(dst16_tmp, dst_rect->w * sizeof(uint16_t));
-        dst16_tmp += dst_img->w;
+    if (dst_img->w == dst_rect->w) {
+        SCB_CleanInvalidateDCache_by_Addr(dst16, dst_rect->w * dst_rect->h * sizeof(uint16_t));
+    } else {
+        uint16_t *dst16_tmp = dst16;
+        for (int i = 0; i < dst_rect->h; i++) {
+            SCB_CleanInvalidateDCache_by_Addr(dst16_tmp, dst_rect->w * sizeof(uint16_t));
+            dst16_tmp += dst_img->w;
+        }
```

Comment:
Why was line-by-line invalidate ever needed?

Reply:
Yes, any operations involving cropping the middle of an image and/or drawing the middle of another image will be faster.

Reply:
And every other normal use case will be slower.